processing/src/main/java/io/druid/query/metadata/metadata/SegmentMetadataQuery.java

```diff
@@ -255,6 +256,14 @@ public boolean hasMinMax()
     return Druids.SegmentMetadataQueryBuilder.copy(this).toInclude(includerator).build();
   }
 
+  public SegmentMetadataQuery withFinalizedAnalysisTypes(SegmentMetadataQueryConfig config)
+  {
+    return Druids.SegmentMetadataQueryBuilder
```

Comment:
Don't create a new query object if analysisTypes are already non-null, just return `this`